instrumentation/active_support/test/opentelemetry/instrumentation/active_support/span_subscriber_test.rb

```diff
@@ -205,123 +205,226 @@ def finish(name, id, payload)
   end
 
   describe 'instrument' do
-    before do
-      ActiveSupport::Notifications.unsubscribe(notification_name)
+    after do
+      ActiveSupport::Notifications.notifier.all_listeners_for(notification_name).each do |listener|
+        ActiveSupport::Notifications.unsubscribe(listener)
+      end
```

Comment:
This prevents subscriptions from leaking across tests. Just using `ActiveSupport::Notifications.unsubscribe` with a `Regexp` won't work as intended.